engine/chain/chain_service.go

```diff
@@ -0,0 +1,18 @@
+package chain
+
+import (
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/internal/query"
+	"github.com/go-resty/resty/v2"
+	"github.com/rs/zerolog"
+)
+
+type chainService struct {
+	QueryService
+}
+
+// NewChainService creates a new chain service.
+func NewChainService(logger zerolog.Logger, arcURL, arcToken, deploymentID string) Service {
+	return &chainService{
+		query.NewQueryService(logger.With().Str("chain", "query").Logger(), resty.New(), arcURL, arcToken, deploymentID),
```

Comment:
Maybe I would do that but please let's postpone it to the following PRs when we'll have more that one service (e.g. two xD)

Reply:
But this will enable us to create a service for merkleroots,
one of the reason we postponed that, it was to prevent reimplementing shared resty for each.

Reply:
ok, ok 👍 Let's do this :)